keyboard.js

```diff
@@ -1704,6 +1705,35 @@ var VKI_attach, VKI_close;
           VKI_mouseEvents(numbkspspan);
           thth.appendChild(numbkspspan);
 
+        if (this.VKI_movement) {
+          function moveAt(pageX, pageY) {
+            this.VKI_keyboard.style.left = pageX + VKI_scrollDist()[0] - keyX + VKI_scrollDist()[0] + 'px';
+            this.VKI_keyboard.style.top = pageY + VKI_scrollDist()[1] - keyY + VKI_scrollDist()[1] + 'px';
+          }
+          function onMouseMove(event) {
+            moveAt(event.pageX, event.pageY);
+          }
+          function mouseUp() {
+            document.removeEventListener('mousemove', onMouseMove);
+            document.removeEventListener('mouseup', mouseUp);
+          }
+          let moveOsk = document.createElement('span');
+            moveOsk.id = 'moveOsk';
+            moveOsk.appendChild(document.createTextNode('\u2725'));
+            moveOsk.title = this.VKI_i18n['14'];
+            moveOsk.addEventListener('mousedown', function(event) {
+                event.preventDefault();
+                cord = document.getElementById('keyboardInputMaster').getBoundingClientRect();
+                keyX = event.pageX + VKI_scrollDist()[0] - cord.left;
+                keyY = event.pageY + VKI_scrollDist()[1] - cord.top;
```

Comment:
There needs to be a fork here based on `self.VKI_target.keyboardPosition == 'fixed'`. If this value is equal to `'fixed'` then scroll distance should be ignored.

Also, these variables should be declared with `let`.